### PR TITLE
Add basic CI

### PR DESCRIPTION
**Pull request by CPerezz**

### CODE REVIEW

File: src/spend/secret.rs [hunk 1]
```diff
@@ -64,15 +64,15 @@ impl SecretKey {
 
     /// Derive the secret to deterministically construct a [`PublicKey`]
     pub fn public_key(&self) -> PublicKey {
-        let A = GENERATOR_EXTENDED * &self.a;
-        let B = GENERATOR_EXTENDED * &self.b;
+        let A = GENERATOR_EXTENDED * self.a;
```
Comment (vlopes11) [inline src/spend/secret.rs:67]: Shouldn't we `allow_non_snake_case`?

File: src/view.rs [hunk 2]
```diff
@@ -53,7 +53,7 @@ impl ViewKey {
 
     /// Derive the secret to deterministically construct a [`PublicSpendKey`]
     pub fn public_key(&self) -> PublicSpendKey {
-        let A = GENERATOR_EXTENDED * &self.a;
+        let A = GENERATOR_EXTENDED * self.a;
```
Comment (vlopes11) [inline src/view.rs:56]: Same as above

**Review by vlopes11 — APPROVED** [review]
LGTM, small nit

File: src/view.rs (hunk 2 above)
Comment (CPerezz) [reply src/view.rs:56]: It's enabled for the whole repo in `lib.rs`

File: src/spend/secret.rs (hunk 1 above)
Comment (CPerezz) [reply src/spend/secret.rs:67]: It's enabled for the whole repo in `lib.rs`

**Pull request merged**